pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2553,6 +2561,35 @@ func handlePhysicalIOAdapterListImpl(ctxArg interface{}, key string,
 			// We assume AddOrUpdateIoBundle will preserve any
 			// existing IsPort/IsPCIBack/UsedByUUID
 			aa.AddOrUpdateIoBundle(log, *ib)
+
+			// Adding PF before VF to have correct boot order
+			if ib.Type == types.PhyIoNetEthPF && ib.Vfs.Count > 0 {
+				ifNameMap, err := nettb.GetPciToIfNameMapByTimeout(60 * time.Second)
+				if err != nil {
+					log.Fatal("Failed to obtain ifNameMap")
+				}
+
+				err = createVF(ifNameMap[ib.PciLong], ib.Vfs.Count)
+				if err != nil {
+					log.Fatal("Failed to create VF for iface with PCI address", ib.PciLong)
+				}
+
+				vfs, err := getVfByTimeout(30*time.Second, ifNameMap[ib.PciLong])
+				if err != nil {
+					log.Fatal("Failed to get VF for iface ", ifNameMap[ib.PciLong], " ", err)
+				}
+				if len(vfs.Data) == 0 {
+					log.Fatal("VF list is empty for iface ", ifNameMap[ib.PciLong])
```

Comment:
Do we really need to use log.Fatal() in this function? Or can we still replace it with another type of informing?

Reply:
If there're no VFs created it means that there will be no Interfaces which user expecting from devmodel file. I think that's a good reason to send fatal message

Reply:
Well, actually this check is redundant, since getVfByTimeout returns non-empty VF list, otherwise it gives error, so I'll remove this line. Thanks! But the discussion about recovery is still valid for comment above

Reply:
The original question was not only about checking in the specified line but about all log.Fatal in this function

Reply:
Can we for example use log.Error() and prevent restart EVE from calling Fatal() by handling the problem and reacting to it differently?

Reply:
This comes down to [this](https://github.com/lf-edge/eve/pull/2828#discussion_r980690411) discussion. If we will not be able to create virtual functions, there will be divergence between device model that user requested and what EVE could create.